### PR TITLE
fixed for implementation of selectively ignore some measure issue#372

### PR DESCRIPTION
**Pull request by DeepshikhaNEC**
Add the feature "selectively ignore some measure"

*Edit*: issue #372

### CODE REVIEW

**Comment by DeepshikhaNEC** [comment]
@fgalan Hi, I have fixed this issue #372 and PR has been raised in iotagent-ul as well as iotagent-node-lib for this issue.

Request you to please confirm the same and if the PR is ok then please merge it into master branch.

File: package.json [hunk 1]
```diff
@@ -38,7 +38,7 @@
     "body-parser": "1.18.3",
     "dateformat": "3.0.3",
     "express": "~4.16.4",
-    "iotagent-node-lib": "git://github.com/telefonicaid/iotagent-node-lib.git#master",
+    "iotagent-node-lib": "git://github.com/DeepshikhaNEC/iotagent-node-lib.git#issue/372",
```
Comment (fgalan) [inline package.json:41]: Looking to sibling PR https://github.com/telefonicaid/iotagent-json/pull/458, it uses:

```
"iotagent-node-lib": "git://github.com/Madhu-NEC/iotagent-node-lib.git#selectively_ignore_some_measure",
```

Do they correspond to the same iotagent-node-lib code?

(Anyway, don't worry about this... first PR https://github.com/telefonicaid/iotagent-node-lib/pull/835 will be merged, then this packages.json restored to master)

File: test/unit/ngsiv2/httpBindings-test.js [hunk 2]
```diff
@@ -786,4 +786,270 @@ describe('HTTP Transport binding: measures', function() {
...
+            // provisioning folder of iotagent-node-lib
+            contextBrokerMock
+                .matchHeader('fiware-service', 'smartGondor')
+                .matchHeader('fiware-servicepath', '/gardens')
+                .post('/v2/entities?options=upsert')
+                .reply(204)
+                .post(
+                    '/v2/entities/TimeInstant%20New%20Device/attrs',
+                    utils.readExampleFile('./test/unit/ngsiv2/contextRequests/timeInstantMeasures2.json')                )
+                .query({ type: 'clock' })
```
Comment (fgalan) [inline test/unit/ngsiv2/httpBindings-test.js:825]: We have:

* First test case:
    * L806: `utils.readExampleFile('./test/unit/ngsiv2/deviceProvisioning/provisionDeviceTimeinstant.json')`
    * L824: `utils.readExampleFile('./test/unit/ngsiv2/contextRequests/timeInstantMeasures2.json')`
* Second test case:
    * L872: `utils.readExampleFile('./test/unit/ngsiv2/deviceProvisioning/provisionDeviceTimeinstant2.json')`
    * L890: `utils.readExampleFile('./test/unit/ngsiv2/contextRequests/timeInstantMeasures.json')`
* Third test case:
    * L938: utils.readExampleFile('./test/unit/ngsiv2/deviceProvisioning/provisionDeviceTimeinstant.json')
    * L956: ` utils.readExampleFile('./test/unit/ngsiv2/contextRequests/timeInstantMeasures2.json')`
* Fourth test case:
    * L1004: `utils.readExampleFile('./test/unit/ngsiv2/deviceProvisioning/provisionDeviceTimeinstant2.json')`
    * L1022: `utils.readExampleFile('./test/unit/ngsiv2/contextRequests/timeInstantMeasures.json')`

Could some of the .json files be renamed so we have aligned pairs in "2" and "no-2":

```
utils.readExampleFile('./test/unit/ngsiv2/deviceProvisioning/provisionDeviceTimeinstant.json')
utils.readExampleFile('./test/unit/ngsiv2/contextRequests/timeInstantMeasures.json')
```

and

```
utils.readExampleFile('./test/unit/ngsiv2/deviceProvisioning/provisionDeviceTimeinstant2.json')
utils.readExampleFile('./test/unit/ngsiv2/contextRequests/timeInstantMeasures2.json')
```

Comment (DeepshikhaNEC) [reply test/unit/ngsiv2/httpBindings-test.js:825]: > We have:
> 
> * First test case:
>   
>   * L806: `utils.readExampleFile('./test/unit/ngsiv2/deviceProvisioning/provisionDeviceTimeinstant.json')`
>   * L824: `utils.readExampleFile('./test/unit/ngsiv2/contextRequests/timeInstantMeasures2.json')`
> * Second test case:
>   
>   * L872: `utils.readExampleFile('./test/unit/ngsiv2/deviceProvisioning/provisionDeviceTimeinstant2.json')`
>   * L890: `utils.readExampleFile('./test/unit/ngsiv2/contextRequests/timeInstantMeasures.json')`
> * Third test case:
>   
>   * L938: utils.readExampleFile('./test/unit/ngsiv2/deviceProvisioning/provisionDeviceTimeinstant.json')
>   * L956: ` utils.readExampleFile('./test/unit/ngsiv2/contextRequests/timeInstantMeasures2.json')`
> * Fourth test case:
>   
>   * L1004: `utils.readExampleFile('./test/unit/ngsiv2/deviceProvisioning/provisionDeviceTimeinstant2.json')`
>   * L1022: `utils.readExampleFile('./test/unit/ngsiv2/contextRequests/timeInstantMeasures.json')`
> 
> Could some of the .json files be renamed so we have aligned pairs in "2" and "no-2":
> 
> ```
> utils.readExampleFile('./test/unit/ngsiv2/deviceProvisioning/provisionDeviceTimeinstant.json')
> utils.readExampleFile('./test/unit/ngsiv2/contextRequests/timeInstantMeasures.json')
> ```
> 
> and
> 
> ```
> utils.readExampleFile('./test/unit/ngsiv2/deviceProvisioning/provisionDeviceTimeinstant2.json')
> utils.readExampleFile('./test/unit/ngsiv2/contextRequests/timeInstantMeasures2.json')
> ```

@fgalan Hi, I have been incorporated the changes suggested by you. Thanks!

File: config.js [hunk 3]
```diff
@@ -341,7 +341,9 @@ config.iota = {
     /**
      * Default type, for IoT Agent installations that won't require preregistration.
      */
-    defaultType: 'Thing'
+    defaultType: 'Thing',
+
+    explicitAttrs: false
```
Comment (fgalan) [inline config.js:346]:
```suggestion
    /**
     * flag indicating whether the incoming measures to the IoTAgent should be processed as per the "attributes" field.
     */
    explicitAttrs: false
```

To mimics how looks like in the sibling PR at IOTA-JSON

**Comment by fgalan** [comment]
Now that https://github.com/telefonicaid/iotagent-node-lib/pull/873 has been merged, packages.json in this PR can be changed to master.

**Comment by DeepshikhaNEC** [comment]
> Now that [telefonicaid/iotagent-node-lib#873](https://github.com/telefonicaid/iotagent-node-lib/pull/873) has been merged, packages.json in this PR can be changed to master.

@fgalan Hi, I have updated the changes as suggested by you. Thanks !

**Pull request merged**